### PR TITLE
Update CI

### LINKED ISSUE #14: Update CI
I believe of all of our projects, `intake-esm` is the most "up-to-date" with regards to CI testing.  I think that `mpirical` should match its CI testing framework.  This means:

1. Updating the CircleCI tests to implement an upstream-dev job, adding tests for Python 3.7 and 3.8, adding nightly builds, etc.
2. Add a Branch Protection Rule (in the GitHub repo settings) to enforce non-upstream-dev tests pass for PRs and that merging directly into `master` is not allowed.
3. Add GitHub Actions for code style enforcement (black, flake8, and isort).

Should contact @andersy005 for help, if necessary.

### PR DESCRIPTION
**Pull request by jukent**
Closes #14

### CODE REVIEW

**Comment by jukent** [comment]
Next week:

Why is 3.8-mpich failing?
Add upstream dev and nightly builds.

**Comment by jukent** [comment]
whichcraft and mpi4py not installing on 3.8 mpich

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,88 +10,80 @@ workflows:
...
 
   test-3.8-mpich:
     <<: *default-job
+    docker:
+      - image: ncarxdev/miniconda:3.8
     environment:
-      ENV_NAME: "mpirical"
-      ENV_SCRIPT: "env-3.8-mpich.yml"
-      TEST_SCRIPT: "test-unit-only.sh"
+      ENV_FILE: ".circleci/env-3.8-mpich.yml"
```
Comment (kmpaul) [inline .circleci/config.yml]: @jukent This is the problem!  You need to change this line to read:

```suggestion
      ENV_FILE: ".circleci/env-mpich.yml"
```

**Review by kmpaul — CHANGES_REQUESTED** [review]
@jukent Note the changes requested.  Just forgot to change the name of the environment file in the `test-3.8-mpich` job.

**Pull request merged**